### PR TITLE
Release candidate - changes required for SDKS.

### PR DESCRIPTION
**Pull request by p-zielinski**
## 2024-07-30

- Added parameter `ParameterRedeemableHolderId`
- Schema `Session` - property `key` is no longer `enum`
- Added `RewardsListResponseBody`, `PromotionsTiersCreateRequestBody`, `PromotionsTiersCreateResponseBody`, `PromotionsTiersUpdateRequestBody`, `PromotionsTiersUpdateResponseBody`, `PromotionTierUpdate`, `PromotionTierCreateParams`, `RewardsCreateRequestBodyMaterial`, `RewardsCreateRequestBodyDigital`, `RewardsCreateRequestBodyDigitalParameters`, `RewardsCreateRequestBodyPayWithPoints`,  `RewardsUpdateRequestBody`, `RewardsUpdateRequestBodyDigital`, `RewardsUpdateRequestBodyPayWithPoints`, `RewardsUpdateRequestBodyMaterial`, `RewardsUpdateRequestBodyDigitalParametersLoyaltyProgram`, `RewardsUpdateRequestBodyDigitalParametersGiftVouchers`, `RewardsUpdateRequestBodyDigitalParametersDiscountCoupons`, `RewardsUpdateRequestBodyDigitalParameters`, `RewardsUpdateRequestBodyMaterialParameters`, `RewardsUpdateRequestBodyPayWithPointsParameters`, `RewardsCreateRequestBodyDigitalParametersLoyaltyProgram`, `RewardsCreateRequestBodyDigitalParametersGiftVouchers`, `RewardsCreateRequestBodyDigitalParametersDiscountCoupons`, `RewardsCreateRequestBodyMaterialParameters`, `RewardsCreateRequestBodyPayWithPointsParameters`,`SegmentsCreateRequestBodyStatic`,  `SegmentsCreateRequestBody`, `SegmentsGetResponseBody`, `SegmentsCreateResponseBody`, `SegmentsCreateRequestBodyDynamic`, `AsyncActionGetResponseBody`, `AsyncAction`, `AsyncActionsListResponseBody`, `ReferralsMembersHoldersCreateInBulkRequestBody`, `ReferralsMembersHoldersCreateInBulkResponseBody`, `RedemptionEntry`
- Deleted `CodeConfigRequiredLengthCharsetPattern`

  `OpenAPI.json` and `OpenAPIWebhooks.json`:
- Added `AccessSettingsCampaignAssignmentsList` and `AreaStoreCampaignAssignment` schemas for the Areas and Stores feature
- Added `access_settings_assignments` field to the `Campaign` schema

- Added POST `v1/referrals/{campaignId}/members/{memberId}/holders` and `v1/referrals/members/{memberId}/holders`
- Added DELETE `v1/referrals/{campaignId}/members/{memberId}/holders/{holderId}` and `v1/referrals/members/{memberId}/holders/{holderId}`
- Added a `holder_role` filter to Qualifications

- added schemas `ValidityTimeframe` and `ValidityDayOfWeek`, replaced everywhere `validity_day_of_week` with ref to `ValidityDayOfWeek`, replaced everywhere `validity_timeframe` with ref to `ValidityTimeframe`

### CODE REVIEW

**Review by p-woznikowski — CHANGES_REQUESTED** [review]
Ogarnąłem `OpenAPI.json`. Zostawiam:

- prośby o drobne zmiany
- sugestie zmian
- pytania

File: reference/OpenAPI.json [hunk 1]
```diff
@@ -9637,6 +9481,145 @@
...
+            "type": "string",
+            "description": "Assign a new or update the promotion tier's category using id",
+            "example": "cat_0c9da30e7116ba6bba"
+          }
+        },
+        "required": [
+          "name"
+        ]
+      },
+      "PromotionTierUpdate": {
```
[the remark this answers is not in the document]
Comment (p-zielinski) [reply reference/OpenAPI.json:9557]: Myślałem nad tym i dla SDK jest lepiej jak są 2 różne schematy, daje to większość dowolność zmian.
Fakt, jak przyjdzie do jakichś zmian to trzeba je robić w paru miejscach, ale czasem tak jest lepiej.

File: reference/OpenAPI.json [hunk 2]
```diff
@@ -10314,57 +10052,42 @@
...
+            "$ref": "#/components/schemas/RewardsUpdateRequestBodyMaterial"
           },
           {
-            "$ref": "#/components/schemas/4_req_update_reward_MATERIAL"
+            "$ref": "#/components/schemas/RewardsUpdateRequestBodyPayWithPoints"
           }
         ]
       },
-      "4_req_update_reward_CAMPAIGN": {
+      "RewardsUpdateRequestBodyDigital": {
```
[the remark this answers is not in the document]
Comment (p-zielinski) [reply reference/OpenAPI.json:10090]: Dorzucę do w kolejnej PR

File: reference/OpenAPI.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (p-woznikowski) [reply reference/OpenAPI.json:9557]: Ok, rozumiem, dzięki

File: reference/OpenAPI.json [hunk 3]
```diff
@@ -30309,31 +29781,7 @@
           },
           "type": {
             "type": "string",
-            "description": "Type of async action.",
-            "enum": [
```
[the remark this answers is not in the document]
Comment (p-zielinski) [reply reference/OpenAPI.json:30313]: ok ogarnę sobie to skryptem, bo w sdk chcemy by to był string

**Review by p-woznikowski — APPROVED** [review]

**Pull request merged**